### PR TITLE
test(e2e): add v4 sse entrypoint to mock endpoint

### PR DESCRIPTION
**Pull request by marcambier**
**Issue**

https://github.com/gravitee-io/issues/issues/8332

**Description**

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/test-v4mockendpoint-run-e2e/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-nhyfudfgjx.chromatic.com)

---
### 🧪 End-to-End Coverage

| INSTRUCTIONS | BRANCHES |
| :----------: | :------: |
|   36% | 20%   |

A more detailed report has been uploaded to [circleci](https://output.circle-artifacts.com/output/job/697791c2-f268-44c1-8a10-6c9621302db1/artifacts/0/gravitee-apim-e2e/jacoco/reports/index.html)

### CODE REVIEW

File: gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    });
+  });
+
+  describe('Gateway call', () => {
+    test('Gateway call should return backend response', async () => {
+      const messages = [];
+      await fetchEventSourceGateway({ contextPath: `/${contextPath}` }, (ev) => {
+        messages.push(ev.data);
+      });
+      expect(messages).toHaveLength(1);
```
Comment (marcambier) [inline gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts]: Why is there only 1 message ? Shouldn't it be 3 according to the endpoint configuration ?

File: gravitee-apim-e2e/lib/utils/eventsource-fetch.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
```
Comment (guillaumelamirand) [inline gravitee-apim-e2e/lib/utils/eventsource-fetch.ts:2]: Those are not copyright from graviteee but comes from microsoft see https://github.com/Azure/fetch-event-source/tree/main/src.

You should refer to those lib or at least give the source somewhere.

File: gravitee-apim-e2e/lib/fixtures/management/PlansV4Faker.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (guillaumelamirand) [inline gravitee-apim-e2e/lib/fixtures/management/PlansV4Faker.ts:1]: Those are not copyright from graviteee but comes from microsoft see https://github.com/Azure/fetch-event-source/tree/main/src.

You should refer to those lib or at least give the source somewhere.

**Review by guillaumelamirand — CHANGES_REQUESTED** [review]

File: gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+import { ApisV4Faker } from '@gravitee/fixtures/management/ApisV4Faker';
+import faker from '@faker-js/faker';
+
+const orgId = 'DEFAULT';
+const envId = 'DEFAULT';
+
+const apisResource = new APIsV4Api(forManagementAsApiUser());
+const apiPlansResource = new APIPlansV4Api(forManagementAsApiUser());
+
+describe('Gateway V4 - SSE entrypoint to mock endpoint', () => {
```
Comment (marcambier) [inline gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts]: Should we use `describeIfJupiter` to run only on jupiter gw ?

File: gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts (hunk 1 above)
Comment (marcambier) [reply gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts]: fixed

File: gravitee-apim-e2e/lib/utils/eventsource-fetch.ts (hunk 2 above)
Comment (marcambier) [reply gravitee-apim-e2e/lib/utils/eventsource-fetch.ts:2]: I added a reference in the top comment of the file

File: gravitee-apim-e2e/lib/fixtures/management/PlansV4Faker.ts (hunk 3 above)
Comment (marcambier) [reply gravitee-apim-e2e/lib/fixtures/management/PlansV4Faker.ts:1]: I added a reference in the top comment of the file

File: gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts (hunk 4 above)
Comment (marcambier) [reply gravitee-apim-e2e/api-test/use-case-test/src/management/v4/v4-sse-entrypoint-to-mock-endpoint.spec.ts]: fixed

**Review by guillaumelamirand — APPROVED** [review]

**Pull request merged**